text/0000-cfg-target.md

```diff
@@ -0,0 +1,99 @@
+- Feature Name: `cfg-target`
+- Start Date: 2020-09-27
+- RFC PR: [rust-lang/rfcs#2991](https://github.com/rust-lang/rfcs/pull/2991)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This proposes a new `cfg`: `target`, which matches the entire target triple
+string (e.g. `arm-unknown-linux-gnueabihf`). This also adds a `CARGO_CFG_TARGET`
+environment variable for parity with other `CARGO_CFG_*` variables.
+
+# Motivation
+[motivation]: #motivation
+
+To `#[cfg]` against a specific target, a `build.rs` script is required to emit a
+custom `cfg` based on the `TARGET` environment variable. Adding a build script
+increases compile time and makes a crate incompatible with certain build
+systems.
+
+Otherwise, all available components would need to be specified separately:
```

Comment:
I vaguely remember that the `target` isn't necessarily the concatenation of the various `target_{foo}` parts. That said, I'm having trouble finding the part of the docs that says this...

Reply:
`x86_64-fuchsia` is an example (it doesn't concatenate `unknown`s into the result).

cc @petrhosek @tmandry

Reply:
Clang/LLVM treats empty components, `none` and `unknown` as equivalent, so for example `x86_64-fuchsia`, `x86_64-unknown-fuchsia` and `x86_64-none-fuchsia` are all considered equivalent and internally normalized to `x86_64-unknown-fuchsia`.

We prefer the shortest spelling for convenience, but if Rust prefers always using normalized triples, we could switch to using `x86_64-unknown-fuchsia`, I'd be fine with that. This wasn't possible in the past when LLVM normalized empty components inconsistently, but that issue has since been resolved.